controllers/event_controller.go

```diff
@@ -150,59 +179,71 @@ func (r *EventReconciler) deletePod(ctx context.Context, podName string) error {
 	return nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *EventReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	go func() {
-		ctx := context.Background()
-		countedFlag := make(map[string]bool)
-		for range time.Tick(time.Second) {
-			func() {
-				r.muFirstPodEventTime.Lock()
-				defer r.muFirstPodEventTime.Unlock()
-				r.muPodCreatedEventTime.Lock()
-				defer r.muPodCreatedEventTime.Unlock()
-				for podName, firstTime := range r.firstPodEventTime {
-					if countedFlag[podName] {
-						continue
+func (p *provisionObserver) Start(ctx context.Context) error {
+	countedFlag := make(map[string]bool)
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+		case <-ctx.Done():
+			return nil
+		}
+
+		func() {
+			p.muFirstPodEventTime.Lock()
+			defer p.muFirstPodEventTime.Unlock()
+			p.muPodCreatedEventTime.Lock()
+			defer p.muPodCreatedEventTime.Unlock()
+
+			for podName, firstTime := range p.firstPodEventTime {
+				if firstTime.Before(time.Now().Add(-p.eventTTL)) {
+					delete(p.firstPodEventTime, podName)
+					delete(countedFlag, podName)
+					delete(p.podCreatedEventTime, podName)
```

Comment:
We need to continue after these deletes because the program will crash at the next `countedFlag` reference.